scipy/spatial/ckdtree/src/distance.h

```diff
@@ -52,35 +51,216 @@ typedef BaseMinkowskiDistP2<PlainDist1D> NonOptimizedMinkowskiDistP2;
  * Measuring distances
  * ===================
  */
-inline double
+
+
+#ifndef CKDTREE_BLAS_DIST
+#define CKDTREE_BLAS_DIST
+
+double
+sqeuclidean_distance_double_blas(const double *u, const double *v, ckdtree_intp_t n);
```

Comment:
I suggest rename double_blas to double_default or double_any_n, as it handles the default branch.

Then you can probably always define the prototype and avoid the conditional compilation here.

Reply:
There is no conditional compilation. It is just a macro guard for the forward declaration.

I can name the function whatever you like. It does not really matter what it is called.

Reply:
It is the forward declaration that I feel unease about.

I thought the motivation to conditionally use a forward declaration was to avoid exposing the sqeuclidean_distance_double_blas symbol (which sounded private) to the 'public' API.

Reply:
Ok, I can change the file into a header and declare the function static.

Reply:
I removed it completely, we don't need it. It is only better in pathologically high dimensional spaces (think >512 dimensions or so).

Reply:
LoL